### PR TITLE
Update SnippetConcept.md

### PR DESCRIPTION
**Pull request by ginalee-dotcom**

### CODE REVIEW

File: docs/SnippetConcept.md [hunk 1]
```diff
@@ -4,10 +4,10 @@
 
 Snippets are a helpful tool when creating templates in the FHIR Converter.
 Leveraging snippets allows you to reference them in multiple templates in the future, preventing you from having to rewrite the same code over again.
-Within the FHIR converter release, there are four types of snippets: **Resource**, **Reference**, **Data Type** and **Code System**.
+Within the FHIR converter release, there are seven types of partial templates: **Resources**, **References**, **Data Type**, **Code Systems**, **Sections**, **Utils** and **Value Set**.
```
Comment (BoyaWu10) [inline docs/SnippetConcept.md:7]: `Partial templates` in handlebars are called `snippets` in dotliquid. We can still use `snippets` here.

File: docs/SnippetConcept.md [hunk 2]
```diff
@@ -93,8 +96,24 @@ Below is an example using this code system mapping.
 "gender":"{{ PID.8.Value | get_property: 'CodeSystem/Gender', 'code' }}",
 ```
 
+## C-CDA Specific Partial Templates
+
+The **Sections**, **Utils** and **Value Set** partial templates are unique to C-CDA.
+
+### Sections
+
+Section templates are used in the C-CDA to FHIR Converter. A CDA document is comprised of sections, each of which contain narrative text and some of which contain structured data elements. Examples of these sections include *Encounters*, *Immunization*, *Procedures* and *Vital Signs*. The section templates map these sections to FHIR resources. Each CDA document template is comprised of section partial templates.
```
Comment (BoyaWu10) [inline docs/SnippetConcept.md:105]: We can use consistent `C-CDA` across the repo instead of `CDA`.

File: docs/SnippetConcept.md [hunk 3]
```diff
@@ -93,8 +96,24 @@ Below is an example using this code system mapping.
...
+
+### Value Set
+
+Value Set partial templates are used in the C-CDA to FHIR Converter. These templates map the C-CDA value sets to FHIR value sets and code systems. These are implemented based on the C-CDA and FHIR specifications. We have released a selection of Value Set partial templates, but not the full list of required value sets for C-CDA.  
+
+### Utils
+
+Utils partial templates are used in the implementation of C-CDA to FHIR we have released. These templates provide utility functions for our implementation. For example, Utils/ResourceTypeFromSection.hbs maps a CDA section to a FHIR resource type based on CDA ID values.
```
Comment (BoyaWu10) [inline docs/SnippetConcept.md:113]: Same for `CDA` here.

File: docs/SnippetConcept.md [hunk 4]
```diff
@@ -71,8 +71,11 @@ Below is the example in the ADT_A01 template, where a reference template is incl
 ```
 {% include 'Reference/Encounter/Diagnosis_Condition' with ID: pv1Id, REF: fullDg1Id -%}
 ```
+## HL7 v2 Specific Partial Template
```
Comment (BoyaWu10) [inline docs/SnippetConcept.md:74]: Same for `partial template` here and following sections.

**Comment by ginalee-dotcom** [comment]
@BoyaWu10 Let me know if I should revise anything else! If it looks ready, please merge it. Thank you!

**Review by BoyaWu10 — APPROVED** [review]

**Comment by BoyaWu10** [comment]
> @BoyaWu10 Let me know if I should revise anything else! If it looks ready, please merge it. Thank you!

Thanks Gina! The document looks good to me and Qiwei has helped to merge it.

**Pull request merged**